File: src/2025-02-27-deepclaude37-rust/go/exercises/practice/bottle-song/bottle_song.go
package bottlesong

import "fmt"

func Recite(startBottles, takeDown int) []string {
	numbers := []struct{ title, lower string }{
		{"No", "no"},
		{"One", "one"},
		{"Two", "two"},
		{"Three", "three"},
		{"Four", "four"},
		{"Five", "five"},
		{"Six", "six"},
		{"Seven", "seven"},
		{"Eight", "eight"},
		{"Nine", "nine"},
		{"Ten", "ten"},
	}

	var result []string

	for i := 0; i < takeDown; i++ {
		current := startBottles - i
		if current < 0 {
			continue
		}

		var verse []string

		// First two lines
		var titleWord string
		if current >= 0 && current < len(numbers) {
			titleWord = numbers[current].title
		} else {
			titleWord = ""
		}

		pluralCurrent := "bottles"
		if current == 1 {
			pluralCurrent = "bottle"
		}
		line := fmt.Sprintf("%s green %s hanging on the wall,", titleWord, pluralCurrent)
		verse = append(verse, line, line)

		// Third line
		verse = append(verse, "And if one green bottle should accidentally fall,")

		// Fourth line
		m := current - 1
		var lowerWord, pluralNext string
		if m < 0 {
			lowerWord = "no"
			pluralNext = "bottles"
		} else {
			if m < len(numbers) {
				lowerWord = numbers[m].lower
			} else {
				lowerWord = ""
			}
			if m == 1 {
				pluralNext = "bottle"
			} else {
				pluralNext = "bottles"
			}
		}
		line4 := fmt.Sprintf("There'll be %s green %s hanging on the wall.", lowerWord, pluralNext)
		verse = append(verse, line4)

		// Add verse to result
		result = append(result, verse...)

		// Add empty line if not last verse
		if i < takeDown-1 {
			result = append(result, "")
		}
	}

	return result
}

File: src/python-2025-02-23-yh-deepclaude-retry/python/exercises/practice/bottle-song/bottle_song.py
def recite(start, take=1):
    number_words = {
        10: 'Ten',
        9: 'Nine',
        8: 'Eight',
        7: 'Seven',
        6: 'Six',
        5: 'Five',
        4: 'Four',
        3: 'Three',
        2: 'Two',
        1: 'One'
    }
    
    def get_verse(n):
        bottle_word = 'bottle' if n == 1 else 'bottles'
        line = f"{number_words[n]} green {bottle_word} hanging on the wall,"
        lines = [line, line]
        lines.append("And if one green bottle should accidentally fall,")
        next_n = n - 1
        if next_n == 0:
            next_str = 'no'
            bottle_word_next = 'bottles'
        else:
            next_str = number_words[next_n].lower()
            bottle_word_next = 'bottle' if next_n == 1 else 'bottles'
        lines.append(f"There'll be {next_str} green {bottle_word_next} hanging on the wall.")
        return '\n'.join(lines)
    
    verses = []
    current = start
    for _ in range(take):
        if current < 1:
            break
        verses.append(get_verse(current))
        current -= 1
    
    return '\n\n'.join(verses).split('\n')
